python/rmm/_lib/memory_resource.pyx

```diff
@@ -329,7 +305,7 @@ cpdef _initialize(
     # Save the current device so we can reset it
     try:
         original_device = get_current_device()
-    except RuntimeError:
+    except CUDARuntimeError as e:
```

Comment:
Can we catch the actual proper error here now?

Reply:
Yes, based on the `status`